src/pubsub/pubsub_index.erl

```diff
@@ -33,34 +33,21 @@
 
 -include("pubsub.hrl").
 
--export([init/3, new/1, free/2]).
+-export([init/0, new/1]).
 
-init(_Host, _ServerHost, _Opts) ->
+init() ->
     mnesia:create_table(pubsub_index,
                         [{disc_copies, [node()]},
                          {attributes, record_info(fields, pubsub_index)}]).
 
 new(Index) ->
     case mnesia:read({pubsub_index, Index}) of
         [I] ->
-            case I#pubsub_index.free of
-                [] ->
-                    Id = I#pubsub_index.last + 1,
-                    mnesia:write(I#pubsub_index{last = Id}),
-                    Id;
-                [Id | Free] ->
-                    mnesia:write(I#pubsub_index{free = Free}), Id
-            end;
+            Id = I#pubsub_index.last + 1,
+            mnesia:write(I#pubsub_index{last = Id}),
+            Id;
         _ ->
             mnesia:write(#pubsub_index{index = Index, last = 1, free = []}),
```

Comment:
Do use `free` field now ?

Reply:
The field `free` is not used now. I left it for 2 reasons:
1. If the record `pubsub_index`'s definition is not changed, dialyzer will complain about wrongly used `free` field if it's not set.
2. If the record definition is changed then we'd have migrate mnesia table schema which I wanted to avoid.

There is way to make dialyzer happy but I still want to leave the `free` field in the record as is.